packages/flutter_tools/lib/src/flx.dart

```diff
@@ -36,11 +36,22 @@ class _Asset {
 }
 
 Iterable<_Asset> _parseAssets(Map manifestDescriptor, String manifestPath) sync* {
-  if (manifestDescriptor == null || !manifestDescriptor.containsKey('assets'))
+  if (manifestDescriptor == null)
     return;
   String basePath = path.dirname(path.absolute(manifestPath));
-  for (String asset in manifestDescriptor['assets'])
-    yield new _Asset(base: basePath, key: asset);
+  if (manifestDescriptor.containsKey('assets')) {
+    for (String asset in manifestDescriptor['assets'])
+      yield new _Asset(base: basePath, key: asset);
+  }
+  if (manifestDescriptor.containsKey('asset-lists')) {
+    for (String assetListPath in manifestDescriptor['asset-lists']) {
+      yield new _Asset(base: basePath, key: assetListPath);
+      String assetList = new File(assetListPath).readAsStringSync();
+      for (String asset in assetList.split(new RegExp("\n+")))
+        if (!asset.isEmpty)
```

Comment:
It's a shame to have to repeat this parsing logic in two places.  Can we centralize the logic in a place that can be used both by this tool and by the flutter package?  One way to do that would be to use a standard format like JSON that already has parsing logic built-in.